CITATION.cff

```diff
@@ -0,0 +1,110 @@
+# This CITATION.cff file was generated with cffinit.
+# Visit https://bit.ly/cffinit to generate yours today!
+
+cff-version: 1.2.0
+title: Nex-generation file format (NGFF)
+message: >-
+  If you use this software, please cite it using the
```

Comment:
> This repo isn't really software.

Updated to "Specification" (but "type" must remain "Software" since the only other option is "Dataset").

> Should you cite this repo if you use ome-zarr-py?

I wouldn't say it's a must.